src/main/java/net/sf/jabref/model/database/BibDatabase.java

```diff
@@ -236,14 +235,19 @@ public int getNumberOfKeyOccurrences(String key) {
         return duplicationChecker.getNumberOfKeyOccurrences(key);
     }
 
+    /**
+     * Sets the given key to the given entry.
+     * If the key is null, the entry field will be cleared.
+     *
+     * @return true, if the entry contains the key, false if not
+     */
     public synchronized boolean setCiteKeyForEntry(BibEntry entry, String key) {
-        String oldKey = entry.getCiteKeyOptional().orElse(null);
         if (key == null) {
             entry.clearField(BibEntry.KEY_FIELD);
         } else {
             entry.setCiteKey(key);
         }
-        return duplicationChecker.checkForDuplicateKeyAndAdd(oldKey, key);
+        return duplicationChecker.checkForDuplicateKeyAndAdd(entry.getCiteKeyOptional(), Optional.ofNullable(key));
```

Comment:
`entry.getCiteKeyOptional()` will not contain the old key now. Both arguments will be identical so will break the function of the call.